src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5701,131 +5700,145 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
-                            lastTokenOfList = null;
-                            return ScanTypeFlags.NotType;
+            do
+            {
+                this.EatToken();
 
-                        case ScanTypeFlags.MustBeType:
-                            // We're currently scanning a possible type-argument list.  But we're
-                            // not sure if this is actually a type argument list, or is maybe some
-                            // complex relational expression with <'s and >'s.  One thing we can
-                            // tell though is that if we have a predefined type (like 'int' or 'string')
-                            // before a comma or > then this is definitely a type argument list. i.e.
-                            // if you have:
-                            // 
-                            //      var v = ImmutableDictionary<int,
-                            //
-                            // then there's no legal interpretation of this as an expression (since a
-                            // standalone predefined type is not a valid simple term.  Contrast that
-                            // with :
-                            //
-                            //  var v = ImmutableDictionary<Int32,
-                            //
-                            // Here this might actually be a relational expression and the comma is meant
-                            // to separate out the variable declarator 'v' from the next variable.
-                            //
-                            // Note: we check if we got 'MustBeType' which triggers for predefined types,
-                            // (int, string, etc.), or array types (Goo[], A<T>[][] etc.), or pointer types
-                            // of things that must be types (int*, void**, etc.).
-                            isDefinitelyTypeArgumentList = DetermineIfDefinitelyTypeArgumentList(isDefinitelyTypeArgumentList);
-                            result = ScanTypeFlags.GenericTypeOrMethod;
-                            break;
+                // Type arguments cannot contain attributes, so if this is an open square, we early out and assume it is not a type argument
+                if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
+                {
+                    greaterThanToken = null;
+                    return ScanTypeFlags.NotType;
+                }
 
-                        // case ScanTypeFlags.TupleType:
-                        // It would be nice if we saw a tuple to state that we definitely had a 
-                        // type argument list.  However, there are cases where this would not be
-                        // true.  For example:
+                if (this.CurrentToken.Kind == SyntaxKind.GreaterThanToken)
+                {
+                    greaterThanToken = EatToken();
+                    return result;
+                }
+
+                switch (this.ScanType(out _))
```

Comment:
this out-param never contributed to the greaterThanToken.  it was always overwritten afterwards.  changed to `out _` to be actively less confusing.